### PR TITLE
Multi transfer operations

### PR DESCRIPTION
**Pull request by zemtsov**

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -24,13 +24,22 @@ require (
...
 )
 
+require (
+	github.com/rogpeppe/go-internal v1.12.0 // indirect
+	github.com/sagikazarmark/locafero v0.6.0 // indirect
+	github.com/sagikazarmark/slog-shim v0.1.0 // indirect
+	github.com/sourcegraph/conc v0.3.0 // indirect
+	golang.org/x/exp v0.0.0-20240719175910-8a7402abbf56 // indirect
+)
+
```
Comment (pfi79) [inline go.mod:42]: This section can be added to the lower section. Let's keep 2 sections

File: proto/service.proto [hunk 2]
```diff
@@ -62,6 +62,30 @@ service API {
     };
   }
 
+  rpc MultiTransferByCustomer(MultiTransferBeginCustomerRequest) returns (TransferStatusResponse){
+    option (google.api.http) = {
+      post: "/v1/multi-transfer/customer"
+      body: "*"
+    };
+    option (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_operation) = {
+      summary: "Multi transfer cross chanel be customer";
```
Comment (pfi79) [inline proto/service.proto:71]: be or by?

File: pkg/chproducer/methods.go [hunk 3]
```diff
@@ -319,7 +319,9 @@ func (h *Handler) fromBatchResponse(ctx context.Context, transferID string) (mod
 
 		for _, transaction := range blocks.Transactions {
 			if (transaction.FuncName == model.TxChannelTransferByCustomer.String() ||
-				transaction.FuncName == model.TxChannelTransferByAdmin.String()) &&
+				transaction.FuncName == model.TxChannelTransferByAdmin.String() ||
+				transaction.FuncName == model.TxChannelMultiTransferByCustomer.String() ||
+				transaction.FuncName == model.TxChannelMultiTransferByAdmin.String()) &&
```
Comment (pfi79) [inline pkg/chproducer/methods.go:324]: maybe this check and the analogous ones below should be put into a function?

File: proto/service.proto (hunk 2 above)
Comment (zemtsov) [reply proto/service.proto]: fixed. It was copied from other endpoints descriptions. Fixed them also

File: go.mod (hunk 1 above)
Comment (zemtsov) [reply go.mod]: fixed

File: pkg/chproducer/methods.go (hunk 3 above)
Comment (zemtsov) [reply pkg/chproducer/methods.go]: new package helpers/methods added. It contains some functions to check a kind of method.

File: test/integration/http/http_multi_test.go [hunk 4]
```diff
@@ -0,0 +1,465 @@
...
+			robotProc.Signal(syscall.SIGTERM)
+			Eventually(robotProc.Wait(), network.EventuallyTimeout).Should(Receive())
+		}
+
+		By("stop channel transfer")
+		if channelTransferProc != nil {
+			channelTransferProc.Signal(syscall.SIGTERM)
+			Eventually(channelTransferProc.Wait(), network.EventuallyTimeout).Should(Receive())
+		}
+	})
```
Comment (pfi79) [inline test/integration/http/http_multi_test.go:228]: where stop redis

Comment (zemtsov) [reply test/integration/http/http_multi_test.go]: fixed. added stopping redis stage to http tests

**Review by pfi79 — APPROVED** [review]

**Pull request merged**